document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -352,6 +388,23 @@ public static String parseNonCompositeFilter(
     return filterString.toString();
   }
 
+  private static StringBuilder prepareFilterStringForInOperator(
+      final String parsedExpression, final Iterable<Object> values, final Builder paramsBuilder) {
+    final StringBuilder filterStringBuilder = new StringBuilder();
+    filterStringBuilder.append("(");
+    final String collect =
+        StreamSupport.stream(values.spliterator(), false)
+            .map(
+                val -> {
+                  paramsBuilder.addObjectParam(val);
+                  return "(jsonb_build_array(" + parsedExpression + ") @> jsonb_build_array(?))";
```

Comment:
Also, what happens in case when the LHS is a Function? It's fine if it's not supported. In case it is not supported, worth calling it out.